test/JDBC/expected/cast_eliminate-vu-verify.out

```diff
@@ -120,3 +120,343 @@ text
+Bitmap Heap Scan on cast_eliminate  (cost=8.32..13.67 rows=2 width=4)
+  Recheck Cond: ((roid = 1) OR (roid = 2))
+  ->  BitmapOr  (cost=8.32..8.32 rows=2 width=0)
+        ->  Bitmap Index Scan on cast_eliminate_pkey  (cost=0.00..4.16 rows=1 width=0)
+              Index Cond: (roid = 1)
+        ->  Bitmap Index Scan on cast_eliminate_pkey  (cost=0.00..4.16 rows=1 width=0)
+              Index Cond: (roid = 2)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 0.518 ms
+~~END~~
+
+
+SELECT 1 AS [C1] FROM cast_eliminate WHERE (CAST(ROID AS BIGINT) = 1) AND (CAST(s_int AS BIGINT) = 2)
+GO
+~~START~~
+text
+Query Text: SELECT 1 AS [C1] FROM cast_eliminate WHERE (CAST(ROID AS BIGINT) = 1) AND (CAST(s_int AS BIGINT) = 2)
+Index Scan using cast_eliminate_pkey on cast_eliminate  (cost=0.15..8.17 rows=1 width=4)
+  Index Cond: (roid = 1)
+  Filter: (s_int = 2)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 0.504 ms
+~~END~~
+
+
+SELECT 1 AS [C1] FROM cast_eliminate WHERE 1 = (CAST(ROID AS BIGINT))
+GO
+~~START~~
+text
+Query Text: SELECT 1 AS [C1] FROM cast_eliminate WHERE 1 = (CAST(ROID AS BIGINT))
+Index Only Scan using cast_eliminate_pkey on cast_eliminate  (cost=0.15..8.17 rows=1 width=4)
+  Index Cond: (roid = 1)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 19.299 ms
+~~END~~
+
+
+SELECT 1 AS [C1] FROM cast_eliminate WHERE CAST(CAST(ROID AS BIGINT) as INT) = 1
+GO
+~~START~~
+text
+Query Text: SELECT 1 AS [C1] FROM cast_eliminate WHERE CAST(CAST(ROID AS BIGINT) as INT) = 1
+Index Only Scan using cast_eliminate_pkey on cast_eliminate  (cost=0.15..8.17 rows=1 width=4)
+  Index Cond: (roid = 1)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 25.956 ms
+~~END~~
+
+
+SELECT 1 AS [C1] FROM cast_eliminate WHERE CAST(s_int AS INT) = 1
+GO
+~~START~~
+text
+Query Text: SELECT 1 AS [C1] FROM cast_eliminate WHERE CAST(s_int AS INT) = 1
+Seq Scan on cast_eliminate  (cost=0.00..34.25 rows=10 width=4)
+  Filter: (s_int = 1)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 0.503 ms
+~~END~~
+
+
+SELECT 1 AS [C1] FROM cast_eliminate WHERE CAST(CAST(s_int AS BIGINT) as INT) = 1
+GO
+~~START~~
+text
+Query Text: SELECT 1 AS [C1] FROM cast_eliminate WHERE CAST(CAST(s_int AS BIGINT) as INT) = 1
+Seq Scan on cast_eliminate  (cost=0.00..34.25 rows=10 width=4)
+  Filter: (s_int = 1)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 0.210 ms
+~~END~~
+
+
+SELECT 1 AS [C1] FROM cast_eliminate WHERE CAST(CAST(CAST(s_int AS BIGINT) as INT) as SMALLINT) = 1
+GO
+~~START~~
+text
+Query Text: SELECT 1 AS [C1] FROM cast_eliminate WHERE CAST(CAST(CAST(s_int AS BIGINT) as INT) as SMALLINT) = 1
+Seq Scan on cast_eliminate  (cost=0.00..34.25 rows=10 width=4)
+  Filter: (s_int = 1)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 25.653 ms
+~~END~~
+
+
+-- Bad case: cannot remove CAST if a column is typecasted into a type with less precision
```

Comment:
Could you add one more negative test case like
```
SELECT 1 AS [C1] FROM cast_eliminate WHERE CAST(CAST(ROID AS numeric) as int) = 1
```
?

Reply:
Added. Keep the CASE as expected.